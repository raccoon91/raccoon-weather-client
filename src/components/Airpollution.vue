<template>
  <div :class="['airpollution-container', color(text)]">
    <div class="value">{{value}}</div>
    <div class="mark">㎍/㎥</div>
    <div class="text">{{text}}</div>
  </div>
</template>

<script>
export default {
  name: 'Airpollution',

  props: {
    type: {
      type: String,
      required: true,
    },
    value: {
      type: Number,
      required: true,
    },
  },

  data() {
    return {
      text: '좋음',
    };
  },

  methods: {
    airpollutionGrade(value) {
      if (this.type === 'pm10') {
        switch (true) {
          case value <= 30:
            this.text = '좋음';
            break;
          case value <= 80:
            this.text = '보통';
            break;
          case value <= 100:
            this.text = '나쁨';
            break;
          default:
            this.text = '매우나쁨';
            break;
        }
      } else {
        switch (true) {
          case value <= 15:
            this.text = '좋음';
            break;
          case value <= 35:
            this.text = '보통';
            break;
          case value <= 75:
            this.text = '나쁨';
            break;
          default:
            this.text = '매우나쁨';
            break;
        }
      }
    },
    color(text) {
      switch (text) {
        case '좋음':
          return 'good';
        case '보통':
          return 'normal';
        case '나쁨':
          return 'bad';
        default:
          return 'very-bad';
      }
    },
  },

  created() {
    this.airpollutionGrade(this.value);
  },
};
</script>

<style lang="scss" scoped>
.airpollution-container {
  display: flex;
  align-items: center;
}

.good {
  color: #0074d9;
}

.normal {
  color: #2ecc40;
}

.bad {
  color: #ff851b;
}

.very-bad {
  color: #ff4136;
}

.mark {
  margin-left: 3px;
  margin-right: 10px;
  font-size: 12px;
}
</style>
