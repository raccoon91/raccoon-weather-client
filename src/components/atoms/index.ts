export * from "./Title";
export * from "./Text";
export * from "./Box";
export * from "./Card";
