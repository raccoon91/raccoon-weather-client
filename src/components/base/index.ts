export * from "./Card";
export * from "./Status";
export * from "./styled";
