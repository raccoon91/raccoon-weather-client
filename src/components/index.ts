export * from "./Card";
export * from "./Status";
export * from "./MapChart";
export * from "./Header";
export * from "./Climates";
export * from "./ClimateChart";
export * from "./GlobalTemperatureChart";
