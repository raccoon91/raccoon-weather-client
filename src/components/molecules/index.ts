export * from "./Navgation";
export * from "./WeatherCard";
