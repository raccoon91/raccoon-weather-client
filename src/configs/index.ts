export * from "./theme";
export * from "./globalStyle";
export * from "./chart";
export * from "./region";
