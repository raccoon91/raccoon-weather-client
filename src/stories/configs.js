export const colorOptions = ["black", "white", "blue", "skyBlue", "gray", "darkGray", "background"];

export const textSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl"];

export const titleSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "6xl"];

export const variantOptions = ["primary", "primary-outline"];

export const weatherOptions = {
  Day: { sky: 1, rainType: 0, date: "2022-01-01 08:00" },
  Night: { sky: 1, rainType: 0, date: "2022-01-01 20:00" },
  CloudDay: { sky: 3, rainType: 0, date: "2022-01-01 08:00" },
  CloudNight: { sky: 3, rainType: 0, date: "2022-01-01 20:00" },
  Cloud: { sky: 2, rainType: 0, date: "2022-01-01 08:00" },
  RainyDay: { sky: 3, rainType: 1, date: "2022-01-01 08:00" },
  RainyNight: { sky: 3, rainType: 1, date: "2022-01-01 20:00" },
  RainDrop: { sky: 3, rainType: 4, date: "2022-01-01 20:00" },
  SnowyDay: { sky: 3, rainType: 2, date: "2022-01-01 08:00" },
  SnowyNight: { sky: 3, rainType: 2, date: "2022-01-01 20:00" },
};
