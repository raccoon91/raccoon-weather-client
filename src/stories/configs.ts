export const colorOptions = ["black", "white", "blue", "skyBlue", "gray", "darkGray", "background"];

export const textSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl"];

export const titleSizeOptions = ["2xs", "xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "6xl"];

export const variantOptions = ["primary", "primary-outline"];

export const weatherOptions: { [key: string]: { sky: number; rainType: number; date: string } } = {
  Day: { sky: 1, rainType: 0, date: "2022-01-01 08:00" },
  Night: { sky: 1, rainType: 0, date: "2022-01-01 20:00" },
  CloudDay: { sky: 3, rainType: 0, date: "2022-01-01 08:00" },
  CloudNight: { sky: 3, rainType: 0, date: "2022-01-01 20:00" },
  Cloud: { sky: 2, rainType: 0, date: "2022-01-01 08:00" },
  RainyDay: { sky: 3, rainType: 1, date: "2022-01-01 08:00" },
  RainyNight: { sky: 3, rainType: 1, date: "2022-01-01 20:00" },
  RainDrop: { sky: 3, rainType: 4, date: "2022-01-01 20:00" },
  SnowyDay: { sky: 3, rainType: 2, date: "2022-01-01 08:00" },
  SnowyNight: { sky: 3, rainType: 2, date: "2022-01-01 20:00" },
};

export const labelsMockData = ["2010", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018", "2019", "2020"];

export const lineChartMockData = [10, 12, 11, 10, 9, 8, 8, 10];

export const barChartMockData = [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000];

export const scatterPlotMockData = [
  [30, 31, 32, 33, 34, 35],
  [34, 35, 36],
  [30, 31, 33, 34, 35, 36],
  [31, 35, 36],
  [31, 33, 34],
  [31, 33, 34],
  [30, 31, 32, 33, 34, 34.5, 35],
  [31, 32, 33, 34, 35],
  [31, 32, 33, 34, 35],
  [31, 32, 33, 33.5, 34, 35],
];

export const gradientLineChartMockData = [-0.4, 0, 0.2, 0.3, -0.2, 0, -0.1, 0.2, 0.5, 0.6];
