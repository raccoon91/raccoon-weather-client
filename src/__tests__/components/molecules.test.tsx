import { ThemeProvider } from "styled-components";
import { theme } from "configs";
import { render } from "@testing-library/react";
import { UnitText } from "components/molecules";
import type { ReactNode } from "react";

const renderComponent = (components: ReactNode) => render(<ThemeProvider theme={theme}>{components}</ThemeProvider>);

describe("molecules", () => {
  describe("<UnitText />", () => {
    test("matches snapshot", () => {
      const screen = renderComponent(<UnitText value="100" unit="%" />);

      expect(screen.container).toMatchSnapshot();
    });

    test("shows the props correctly", () => {
      const screen = renderComponent(<UnitText value="100" unit="%" />);
      const text = screen.getByText("100");
      const unit = screen.getByText("%");

      expect(text).toBeInTheDocument();
      expect(unit).toBeInTheDocument();
      expect(text).toHaveStyle("color: #000000");
      expect(unit).toHaveStyle("color: #000000");
    });
  });
});
