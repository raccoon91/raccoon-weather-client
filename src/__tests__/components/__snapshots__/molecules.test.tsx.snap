// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules <UnitText /> matches snapshot 1`] = `
<div>
  <div
    class="sc-dkPtRN gqONxx"
  >
    <h4
      class="sc-iCfMLu QtPJI"
      color="black"
    >
      100
    </h4>
    <p
      class="sc-jrQzAO idwZDg"
      color="black"
    >
      %
    </p>
  </div>
</div>
`;
